### PR TITLE
Rail Blog: Add form

### PR DESCRIPTION
**Pull request by Profsain**
In this part of Rails blog project, I implemented
- New post create a form
- Adding a new comment
- Liking a post

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -5,5 +5,9 @@
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
+
+  resources :posts, only: [:index, :show, :new, :create]
```
Comment (DammyShittu) [inline config/routes.rb]: - [ ] It is redundant to have this definition this way. You have defined post `index` and `show` actions on line 6 yet you are defining them again on line 9. Kindly remove the second one.

File: config/routes.rb [hunk 2]
```diff
@@ -5,5 +5,9 @@
...
+  resources :comments, only: [:new, :create]
+  resources :likes, only: [:create]
```
Comment (DammyShittu) [inline config/routes.rb]: - [ ] We can improve these routes definitions by nesting them. Kindly do that.

  ```
  resources :posts, only: [:new, :create] do
    resources :comments, only: [:new, :create]
    resources :likes, only: [:create]
  end
  ```

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
+    @user = current_user
+    @post.user = @user
+
+    if @post.save
+      flash[:notice] = 'Post was successfully created.'
+      redirect_to user_posts_path(@user)
+    else
+      redirect_to new_post_path
+    end
   end
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb]: - [ ] I am unable to create a post. If you check the errors related to `@post` you would see that comments and likes counters are not defined so the validation failed. You need to implement both counters before attempting to save.

  ```
  @post.likes_counter = 0
  @post.comments_counter = 0
  ```

  image

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.user = current_user
+
+    @comment.save if @comment.valid?
+    redirect_to post_path(@comment.post)
```
Comment (DammyShittu) [inline app/controllers/comments_controller.rb]: - [ ] I get this error after creating a comment. The comment is actually created but your redirection is incorrect. It should be `posts_path(...)`, not `post_path(...)`.

  rurir

File: app/controllers/likes_controller.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class LikesController < ApplicationController
+  def create
+    @like = Like.new(like_params)
+    @like.user = current_user
+
+    @like.save if @like.valid?
+    redirect_to post_path @like.post
```
Comment (DammyShittu) [inline app/controllers/likes_controller.rb]: - [ ] I get this error after creating a like. The like is actually created but your redirection is incorrect. It should be `posts_path(...)`, not `post_path(...)`.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Profsain,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (Profsain) [reply app/controllers/posts_controller.rb]: Thank you for the feedback.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Profsain, 
## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**